server/chat-commands/core.ts

```diff
@@ -1324,28 +1369,30 @@ exports.commands = {
 				}));
 				return false;
 			}
-			let roomList = {};
+			let roomList: any = {};
+			// has to be disabled because roomid is reassigned, but only in an if statement.
+			// eslint-disable-next-line prefer-const
 			for (let roomid of targetUser.inRooms) {
 				if (roomid === 'global') continue;
-				let targetRoom = Rooms.get(roomid);
+				const targetRoom = Rooms.get(roomid);
 				if (!targetRoom) continue; // shouldn't happen
-				let roomData = {};
+				const roomData: AnyObject = {};
 				if (targetRoom.isPrivate) {
 					if (!user.inRooms.has(roomid) && !user.games.has(roomid)) continue;
 					roomData.isPrivate = true;
 				}
 				if (targetRoom.battle) {
-					let battle = targetRoom.battle;
+					const battle = targetRoom.battle;
 					roomData.p1 = battle.p1 ? ' ' + battle.p1.name : '';
 					roomData.p2 = battle.p2 ? ' ' + battle.p2.name : '';
 				}
 				if (targetRoom.auth && targetUser.id in targetRoom.auth) {
-					roomid = targetRoom.auth[targetUser.id] + roomid;
+					roomid = (targetRoom.auth[targetUser.id] + roomid) as RoomID;
```

Comment:
This is not a `RoomID`. You might want to define a `roomname` variable and assign this value to. That will also enable you to remove the ESLint diable.